src/components/PageHeadingFlowRun.vue

```diff
@@ -1,26 +1,77 @@
 <template>
-  <page-heading v-if="flowRun" class="page-heading-flow-run" :crumbs="crumbs">
-    <template #after-crumbs>
+  <div v-if="flowRun" class="page-heading-flow-run">
+    <page-heading class="page-heading-flow-run__heading" :crumbs="crumbs">
+      <template #after-crumbs>
+        <div v-if="flowRun.tags && flowRun.tags.length > 0" class="state-list-item__tags">
+          <p-tag-wrapper v-bind="{ tags: flowRun.tags }" />
+        </div>
+      </template>
+      <template #actions>
+        <template v-if="media.sm">
+          <FlowRunPauseButton :flow-run="flowRun" />
+          <FlowRunResumeButton :flow-run="flowRun" />
+          <FlowRunRetryButton :flow-run="flowRun" />
+          <FlowRunCancelButton :flow-run="flowRun" />
+        </template>
+        <FlowRunMenu :flow-run-id="flowRun.id" :show-all="!media.sm" @delete="emit('delete')" />
+      </template>
+    </page-heading>
+    <div class="page-heading-flow-run__meta">
       <StateBadge :state="flowRun.state" />
-    </template>
-    <template #actions>
-      <template v-if="media.sm">
-        <FlowRunPauseButton :flow-run="flowRun" />
-        <FlowRunResumeButton :flow-run="flowRun" />
-        <FlowRunRetryButton :flow-run="flowRun" />
-        <FlowRunCancelButton :flow-run="flowRun" />
+      <FlowRunStartTime :flow-run="flowRun" />
+      <DurationIconText :duration="flowRun.duration" />
+      <template v-if="!isPending">
+        <FlowRunTaskCount :tasks-count="taskRunsCount">
+          <template #default="{ count }">
+            {{ count }} task {{ toPluralString('run', count) }}
+          </template>
+        </FlowRunTaskCount>
+      </template>
+    </div>
+    <div class="page-heading-flow-run__relationships">
+      <template v-if="flowRun.deploymentId && deployment?.name">
+        <div class="flow-run-list-item__relation">
+          <span>Deployment</span> <DeploymentIconText :deployment-id="flowRun.deploymentId" />
+        </div>
+      </template>
+
+      <template v-if="flowRun.workPoolName">
+        <div class="flow-run-list-item__relation">
+          <span>Work Pool</span> <WorkPoolIconText :work-pool-name="flowRun.workPoolName" />
+        </div>
+      </template>
+
+      <template v-if="flowRun.workQueueName">
+        <div class="flow-run-list-item__relation">
+          <span>Work Queue</span> <WorkQueueIconText :work-queue-name="flowRun.workQueueName" :work-pool-name="flowRun.workPoolName" />
+        </div>
       </template>
```

Comment:
Might be a good time to abstract these into separate `FlowRunDeployment`, `FlowRunWorkPool`, and `FlowRunWorkQueue` components. And share between this and the list item component.

Reply:
Good looking out on this!